posthog/temporal/common/logger.py

```diff
@@ -342,6 +430,58 @@ def mark_queue_done(self, _=None):
         self.queue.task_done()
 
 
+class KafkaLogProducerFromQueueSync:
+    """Produce log messages to Kafka by getting them from a queue."""
+
+    def __init__(
+        self,
+        queue: sync_queue.Queue,
+        shutdown_signal: threading.Event,
+        topic: str = KAFKA_LOG_ENTRIES,
+        key: str | None = None,
+        producer=None,
+    ):
+        self.queue = queue
+        self.shutdown_signal = shutdown_signal
+        self.topic = topic
+        self.key = key
+        self.producer = producer or KafkaProducer(
+            bootstrap_servers=settings.KAFKA_HOSTS,
+            security_protocol=settings.KAFKA_SECURITY_PROTOCOL or "PLAINTEXT",
+            acks="all",
+            api_version=(2, 5, 0),
+            ssl_context=configure_default_ssl_context() if settings.KAFKA_SECURITY_PROTOCOL == "SSL" else None,
+        )
+        self.logger = structlog.get_logger()
+
+    def listen(self):
+        """Listen to messages in the queue and produce them to Kafka."""
+        try:
+            while not self.shutdown_signal.is_set():
```

Comment:
When we receive the signal, we want to make sure we empty the queue as the signal may arrive while there are still logs to process.

Reply:
That could be achieved in the waiting task by first `join`-ing on the queue.